src/test/groovy/no/skatteetaten/aurora/boober/service/AbstractSpec.groovy

```diff
@@ -1,12 +1,21 @@
 package no.skatteetaten.aurora.boober.service
 
+import no.skatteetaten.aurora.boober.Boober
+import no.skatteetaten.aurora.boober.mapper.platform.JavaPlatformHandler
+import no.skatteetaten.aurora.boober.mapper.platform.WebPlatformHandler
+import no.skatteetaten.aurora.boober.platform.ApplicationPlatformHandler
 import com.fasterxml.jackson.databind.JsonNode
 import com.fasterxml.jackson.databind.ObjectMapper
 
 import spock.lang.Specification
 
 abstract class AbstractSpec extends Specification {
```

Comment:
This name was not changed in this PR, but I do find it too general and maybe worth renaming?
AbstractSpec tells me nothing of what the class actually provides. It should reflect that it helps with the loading of resources.

Reply:
+1 from me, almost all specs we write extend this one so naming it BooberSpec could also work. This is not a task for this PR however, so create a new task for it?

Reply:
I find that usually in a project/module one ends up with a couple of loosely related utility functions for testes that are general enough to be relevant for most tests in the project, but does not warrant their own separate utility library. Some of these utility functions are less useful if they are put utility classes (for instance if they are based on conventions using the current test class for naming) and, hence, end up in an abstract class that most tests use.

Though I agree on the basic premise that classes should be named from what they represent or what functionality they provide, there are a few circumstances where this actually is counter productive in my opinion, and this is one of those cases. We could name it AbstractResourceLoadingSpec, for instance, but what if we add a utility method for doing some logging setup? We should probably rename the class, but to what? Or, maybe we should not add the logging utility method to AbstractResourceLoadingSpec at all? Create a new properly named abstract spec class for it and create a hierarchy? Which is more general AbstractLoggingSomethingSpec or AbstractResourceLoadingSpec? Maybe not use abstract classes at all? But then the utility methods becomes more cumbersome to call, and less useful.

If we create an AbstractSpec we can put our most general test utility methods there and not let naming force a counter productive design. Ok, you can't tell from AbstractSpec what it does, but it is easy to find out. And because this is an exception to the general guidelines for class names, it will not become a pattern for bad practice other places in the code. And if we look elsewhere, exactly what does Specification tell us about what functionality that class provides? Nothing about mocking. Or TestCase? Nothing about providing a set of assertion methods. SimpleControllerHandlerAdapter? Is BooberSpec actually better? To me it indicates testing of boober or some boober specific testing functionality. Is that actually correct, or misinforming?

I like AbstractSpec. There should probably be one in every project, containing the most general utilities for tests in that project. It is probably a class that will break the single responsibility principle. But it will keep things DRY, and it will keep things simple. But nobody will be able to tell by its name what it does - only that it is useful and you should probably get to know it.

Reply:
IMHO BooberSpec or MokeySpec gives a clearer intent of what it is then AbstractSpec, but for me this is not a very big deal. Using a spec class that is the base for all our specifications is in general a thing i support.

Reply:
I think you have valid arguments @bentsolheim  
I will not create a big discussion on this topic, its not a big deal if we keep it as-is in this case, but here are my thoughts.

I think really really hard before using inheritance because its the highest level of coupling we can have. If we create a base class that all test classes should extend, we should be very certain of what it contains. I have seen in other projects that these classes have a tendancy to grow out of control. I have experience with all tests becoming slow because of the contents in such a common base class.

The Specification class from Spock I think is a good name. It says to me that it contains all the details of a testcase (specification). So I would actually turn it around and say that its an example of a good name. I dont get all the implementation details from the name, but I do get an indication of its domain. I am not a big fan of naming things unclear because we might sometime in the future expand it. I would rather name it as precise as I can now, and if I need to expand it in the future I will rename it.

The general feeling I have is that putting unrelated features into this very vaguely defined class needs some careful thinking. Maybe its worth the cost in this case, but all I am saying is that we should think twice before adding some new feature into this class. In some cases it might not be worth the cost.

Reply:
Yes, I agree completely with every point you make here, @jarlehansen (I have never seen these classes grow out of control, though - usually only a few methods pass the criteria to be put there).

I'm not a fan either with naming things vaguely or based on potential future usecases (and also cautious with using inheritance). If you take a look at the other abstract test classes they are more clearly named after what functionality they provide. Its just that having a single class between our test classes and whatever test framework we use have pretty much always proved useful to me. AbstractSpec is "our" Specification class. The Specification class we should use.

But, just to be clear. This is an exception to "the rules".